### PR TITLE
feat: password hint

### LINKED ISSUE #118: [UI/UX/API] Account sign up failure when password is not strong enough
While signing up a new account, and the password is not strong enough (for example, it does not contain a special character), sign up just fails with a general error. The response of account creation simply returns with 400 and a general error complaining password field has an issue, without specifying what the issue is.

What is expected
- UI can create in-place checking for the password rule for clearer indication.

### PR DESCRIPTION
**Pull request by bouassaba**
- feat(ui): password hint.
- fix(idp): send proper user message in error.

<br/>
<br/>

<br/>
<br/>

<br/>
<br/>

Closes #118

### CODE REVIEW

**Review by Mrkazik99 — APPROVED** [review]

**Review by loboda4450 — DISMISSED** [review]

File: ui/src/components/sign-up/password-hints.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      <PasswordRequirement
+        text={`Contains at least ${requirements.minUppercase} uppercase character.`}
+        isFulfilled={hasMinUpperCase(value, requirements.minUppercase)}
+      />
+      <PasswordRequirement
+        text={`Contains at least ${requirements.minNumbers} number.`}
+        isFulfilled={hasMinNumbers(value, requirements.minNumbers)}
+      />
+      <PasswordRequirement
+        text={`Contains at least ${requirements.minSymbols} symbol(s).`}
```
Comment (Mrkazik99) [inline ui/src/components/sign-up/password-hints.tsx:30]: I think that more clear for users will be to specify what symbol(s) means.
Example from my previous project
image

**Review by Mrkazik99 — CHANGES_REQUESTED** [review]

File: ui/src/components/sign-up/password-hints.tsx (hunk 1 above)
Comment (loboda4450) [reply ui/src/components/sign-up/password-hints.tsx:30]: And special character is more precise than symbol i guess

File: idp/.env [hunk 2]
```diff
@@ -11,6 +11,13 @@ TOKEN_ISSUER="localhost"
 TOKEN_ACCESS_TOKEN_LIFETIME=86400
 TOKEN_REFRESH_TOKEN_LIFETIME=2592000
 
+# Password
+PASSWORD_MIN_LENGTH=8
+PASSWORD_MIN_LOWERCASE=1
+PASSWORD_MIN_UPPERCASE=1
+PASSWORD_MIN_NUMBERS=1
+PASSWORD_MIN_NUMBERS=1
```
Comment (dsonck92) [inline idp/.env:19]:
```suggestion
PASSWORD_MIN_SYMBOLS=1
```

File: idp/src/config/config.ts [hunk 3]
```diff
@@ -46,6 +47,14 @@ export function readToken(config: Config) {
   }
 }
 
+export function readPassword(config: Config) {
+  config.password.minLength = parseInt(process.env.PASSWORD_MIN_LENGTH)
+  config.password.minLowercase = parseInt(process.env.PASSWORD_MIN_LOWERCASE)
+  config.password.minUppercase = parseInt(process.env.PASSWORD_MIN_UPPERCASE)
+  config.password.minNumbers = parseInt(process.env.PASSWORD_MIN_NUMBERS)
+  config.password.minSymbols = parseInt(process.env.PASSWORD_MIN_NUMBERS)
```
Comment (dsonck92) [inline idp/src/config/config.ts:55]:
```suggestion
  config.password.minSymbols = parseInt(process.env.PASSWORD_MIN_SYMBOLS)
```

**Review by dsonck92 — CHANGES_REQUESTED** [review]

File: ui/src/components/sign-up/password-hints.tsx (hunk 1 above)
Comment (bouassaba) [reply ui/src/components/sign-up/password-hints.tsx]: Done.

File: idp/src/config/config.ts (hunk 3 above)
Comment (bouassaba) [reply idp/src/config/config.ts]: Done.

**Review by Mrkazik99 — APPROVED** [review]

**Review by dsonck92 — APPROVED** [review]

**Pull request merged**